usaspending_api/etl/management/commands/elasticsearch_indexer.py

```diff
@@ -0,0 +1,234 @@
+import logging
+
+from datetime import datetime, timezone
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from pathlib import Path
+from time import perf_counter
+
+from usaspending_api.broker.helpers.last_load_date import get_last_load_date
+from usaspending_api.common.elasticsearch.client import instantiate_elasticsearch_client
+from usaspending_api.common.elasticsearch.elasticsearch_sql_helpers import ensure_view_exists
+from usaspending_api.common.helpers.date_helper import datetime_command_line_argument_type, fy as parse_fiscal_year
+from usaspending_api.common.helpers.fiscal_year_helpers import create_fiscal_year_list
+from usaspending_api.etl.elasticsearch_loader_helpers.es_etl_helpers import format_log
+from usaspending_api.etl.elasticsearch_loader_helpers.rapidloader import Rapidloader
+
+logger = logging.getLogger("script")
+
+
+class Command(BaseCommand):
+    """ETL script for indexing transaction data into Elasticsearch
+
+    HIGHLEVEL PROCESS OVERVIEW
+         1. Generate the full list of fiscal years to process as jobs
+         2. Iterate by job
+           a. Download a CSV file by year (one at a time)
+               i. Continue to download a CSV file until all years are downloaded
+           b. Upload a CSV to Elasticsearch
+               i. Continue to upload a CSV file until all years are uploaded to ES
+           c. Delete CSV file
+    TO RELOAD ALL data:
+        python3 manage.py es_rapidloader --index-name <NEW-INDEX-NAME> --create-new-index all
+
+        Running with --new-index will trigger several actions:
+        0. A view will be created in the source database for the ETL queries
+        1. A new index will be created from the value provided by --index-name (obviously)
+        2. A new index template will be loaded into the cluster to set mapping and index metadata
+        3. All aliases used by the API queries will be re-assigned to the new index
+        4. An alias for incremental indexes will be applied to the new index
+        5. If any previous indexes existed with the API aliases, they will be deleted.
+    """
```

Comment:
Command name in docs needs updating. If this is out of date, probably best to review rest of docs to ensure it is accurately describing process / usage. You could move this to the django command help text instead.

Reply:
Agree 100%. This PR was only to reorganize code with a few modifications as possible. The GH U/I did a poor job showing that.